WordPress/gutenberg
# SelectControl: Add flag for larger default size

Part of #39397
In preparation for #41973

## What?

Adds the `__next36pxDefaultSize` prop to SelectControl for the larger default size.

## Why?

The #39397 project is on hold, but this one is necessary for implementing size variants for CustomSelectControl (#42460). Basically, I want both SelectControl and CustomSelectControl to share the underlying styling components, and CustomSelectControl already has a `__next36pxDefaultSize` flag. So we need to implement the flag on SelectControl for parity.

## Testing Instructions

1. `npm run storybook:dev` and see the SelectControl story.
2. `__next36pxDefaultSize` should only affect the `default` size. Other size variants should be unaffected.
